nx.json

```diff
@@ -0,0 +1,43 @@
+{
+  "tasksRunnerOptions": {
+    "default": {
+      "runner": "nx/tasks-runners/default",
+      "options": {
+        "cacheableOperations": [
+          "compile",
+          "lint",
+          "version:update"
+        ]
+      }
+    }
+  },
+  "targetDefaults": {
+    "compile": {
+      "dependsOn": [
+        "version:update",
+        "^compile"
```

Comment:
note for reviewer: this is the syntax to tell that dependencies within the monorepo should be compiled before compiling the current package